pkg/controllers/dnspolicy/dnspolicy_dnsrecords.go

```diff
@@ -216,11 +219,10 @@ func getClusterGatewayAddresses(gw *gatewayv1beta1.Gateway) map[string][]gateway
 	return clusterAddrs
 }
 
-func listenerTotalAttachedRoutes(upstreamGateway *gatewayv1beta1.Gateway, downstreamCluster string, specListener gatewayv1beta1.Listener, addresses []gatewayv1beta1.GatewayAddress) int {
```

Comment:
Could listenerTotalAttachedRoutes be moved to the wrapper?

Reply:
moved. Thanks for spotting!